### PR TITLE
Update Authorization to return state

### PR DESCRIPTION
**Pull request by wusgarden**
I realized that I need the Authorization to return state from requests so I can ensure that the response maps correctly to the request I sent.  Please let me know if you need me to create/update any of the existing unit tests

### CODE REVIEW

**Comment by rossitsaborissova** [comment]
Thanks for your pull request. It's Ok. Please, just add a unit test that checks whether the state is returned in the authorization response if it presents in the request. You can see the following test when_issue_auth_code_check_scope_valid in com.apifest.oauth20.AuthorizationServerTest for reference.

**Comment by wusgarden** [comment]
@rossitsaborissova - I've added an additional test for verifying that the AuthorizationServer class does return state if it's in the Request. I also checked the AuthRequestTest and it already had the unit tests that I would've added.  I ran them locally and they all passed.  Please let me know if you would like any other additional tests.

File: apifest-oauth20/src/test/java/com/apifest/oauth20/AuthorizationServerTest.java [hunk 1]
```diff
@@ -1097,15 +1100,16 @@ public void when_issue_auth_code_check_scope_valid() throws Exception {
...
         willReturn("basic").given(authServer.scopeService).getValidScope(null, clientId);
 
         // WHEN
-        authServer.issueAuthorizationCode(req);
+        String response = authServer.issueAuthorizationCode(req);
 
         // THEN
         verify(authServer).generateCode();
         verify(authServer.scopeService).getValidScope(null, clientId);
+
```
Comment (rossitsaborissova) [inline apifest-oauth20/src/test/java/com/apifest/oauth20/AuthorizationServerTest.java]: Please, add a check that verifies the state is returned in the response. Also, it is better to have a separate test that checks exactly that.

**Comment by wusgarden** [comment]
Updated the unit test.  I did a squash and I think I may the changes may have been skipped.

**Comment by wusgarden** [comment]
For some reason my unit test is failing.  I've modified it

**Pull request merged**